lib/open_api/docs/doc_v2.rb

```diff
@@ -39,6 +51,44 @@ def routes
           end
         end
       end
+
+      private
+
+      def substitute_references(object)
+        if object.kind_of?(Array)
+          object.collect { |i| substitute_references(i) }
+        elsif object.kind_of?(Hash)
+          return fetch_ref_value(object["$ref"]) if object.keys == ["$ref"]
+          object.each_with_object({}) do |(k, v), h|
+            h[k] = substitute_references(v)
+          end
+        else
+          object
+        end
+      end
+
+      def fetch_ref_value(ref_path)
+        _, section, property = ref_path.split("/")
+        send(section)[property]
+      end
+
+      def substitute_regexes(object)
+        if object.kind_of?(Array)
+          object.collect { |i| substitute_regexes(i) }
+        elsif object.kind_of?(Hash)
+          object.each_with_object({}) do |(k, v), h|
+            h[k] = k == "pattern" ? regexp_from_pattern(v) : substitute_regexes(v)
+          end
+        else
+          object
+        end
+      end
+
+      def regexp_from_pattern(pattern)
+        match = pattern.match(/\A\/(.*)\/\z/)
```

Comment:
The \A feels like it's checked wrong...I think the regex is supposed to be  `/\\\A\/(.*)\/\z/`.  You may also want to anchor it with actual \A and \z

Reply:
Sorry, I don't understand the use case of the extra `\\`.  Wouldn't that try to match a `\` before the start of the line?

I'm expecting the input `pattern` to be something like `/^d+$/` and in that case I want to capture only the `^d+$`.  So I was looking for start of string, then `/`, then capture everything until the next `/`, then the end of the string.

Reply:
As discussed, if we assume a pattern will always be a regex and that it always must start with / and end with /, then let's just check that and then blow up if it's wrong.  Perfer start_with and end_woth aso, for readability.